stmhal/dma.c

```diff
@@ -33,9 +33,149 @@
 #include "py/obj.h"
 #include "irq.h"
 
+
+typedef enum {
+    dma_id_not_defined=-1,
+    dma_id_0,
+    dma_id_1,
+    dma_id_2,
+    dma_id_3,
+    dma_id_4,
+    dma_id_5,
+    dma_id_6,
+    dma_id_7,
+    dma_id_8,
+    dma_id_9,
+    dma_id_10,
+    dma_id_11,
+    dma_id_12,
+    dma_id_13,
+    dma_id_14,
+    dma_id_15,
+} dma_id_t;
+
+typedef struct _dma_descr_t {
+#if defined(MCU_SERIES_F4) || defined(MCU_SERIES_F7)
+    DMA_Stream_TypeDef  *instance;
+#elif defined(MCU_SERIES_L4)
+    DMA_Channel_TypeDef *instance;
+#else
+#error "Unsupported Processor"
+#endif
+    uint32_t        sub_instance;
+    uint32_t        transfer_direction; /* Transfer direction Periphery to memory or vis-versa */
+    dma_id_t        id;
+    const DMA_InitTypeDef *init;
+} dma_descr_t;
+
 #define NSTREAMS_PER_CONTROLLER_LOG2 (3)
-#define NSTREAMS_PER_CONTROLLER (1 << NSTREAMS_PER_CONTROLLER_LOG2)
 #define NCONTROLLERS            (2)
+
+#define DMA_TX_TRANSFER      DMA_MEMORY_TO_PERIPH
+#define DMA_RX_TRANSFER      DMA_PERIPH_TO_MEMORY
```

Comment:
I defined theses macros to make the transfer of the information in the datasheet less error-prone. In the data sheet (RM0351, rev 2, Page 309 ff) there is always a SPI2_TX and SPI2_RX.. So I though it would be a good thing to define macros which are close to what is written in the datasheet.